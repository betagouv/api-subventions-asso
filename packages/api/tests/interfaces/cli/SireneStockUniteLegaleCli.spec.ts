import path from "path";
import fs from "fs";
import sireneStockUniteLegaleApiPort from "../../../src/dataProviders/api/sirene/sireneStockUniteLegale.port";
import sireneStockUniteLegaleDbPort from "../../../src/dataProviders/db/sirene/stockUniteLegale/sireneStockUniteLegale.port";
import SireneStockUniteLegaleCli from "../../../src/interfaces/cli/SireneStockUniteLegale.cli";
import { ObjectId } from "mongodb";
import uniteLegalNamePort from "../../../src/dataProviders/db/uniteLegalName/uniteLegalName.port";
import uniteLegalEntreprisePort from "../../../src/dataProviders/db/uniteLegalEntreprise/uniteLegalEntreprise.port";

let ZIP_PATH = path.resolve(__dirname, "../../../src/modules/providers/sirene/__fixtures__");

describe("SireneStockUniteLegaleCli", () => {
    let getZipMock: jest.SpyInstance;
    beforeAll(() => {
        getZipMock = jest.spyOn(sireneStockUniteLegaleApiPort, "getZip").mockImplementation(() =>
            Promise.resolve({
                data: fs.createReadStream(ZIP_PATH + "/StockUniteLegale_utf8.zip"),
                status: 200,
                statusText: "OK",
            }),
        );
    });

    afterAll(() => {
        jest.restoreAllMocks();
    });

    let cli = new SireneStockUniteLegaleCli();
    describe("parse", () => {
        it("should persist sirene data", async () => {
            await cli.getAndParse();
            // @ts-expect-error: access protected for test
            const data = (await sireneStockUniteLegaleDbPort.collection.find({}).toArray()).map(object => ({
                ...object,
                _id: expect.any(ObjectId),
            }));
            expect(data).toMatchSnapshot();
        });

        it("should persist asso names", async () => {
            await cli.getAndParse();
            // @ts-expect-error: access protected for test
            const data = (await uniteLegalNamePort.collection.find({}).toArray()).map(object => ({
                ...object,
                _id: expect.any(ObjectId),
            }));
            expect(data).toMatchSnapshot();
        });

        it("should persist entreprises' siret", async () => {
            await cli.getAndParse();
            // @ts-expect-error: access protected for test
            const data = (await uniteLegalEntreprisePort.collection.find({}).toArray()).map(object => ({
                ...object,
                _id: expect.any(ObjectId),
            }));
            expect(data).toMatchSnapshot();
        });
    });
});
