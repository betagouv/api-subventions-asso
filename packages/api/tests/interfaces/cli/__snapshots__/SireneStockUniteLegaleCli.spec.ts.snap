// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SireneStockUniteLegaleCli parse should persist asso names 1`] = `
Array [
  Object {
    "_id": Any<ObjectId>,
    "name": "CENTRE MEDICO PSYCHO PEDAGOGIQUE",
    "searchKey": "300483146 - centre medico psycho pedagogique",
    "siren": "300483146",
    "updatedDate": 2012-08-01T00:00:00.000Z,
  },
  Object {
    "_id": Any<ObjectId>,
    "name": "EDUCATION POPULAIRE DU FAUBOURG VOSGES",
    "searchKey": "300846177 - education populaire du faubourg vosges",
    "siren": "300846177",
    "updatedDate": 2008-01-01T00:00:00.000Z,
  },
]
`;

exports[`SireneStockUniteLegaleCli parse should persist entreprises' siret 1`] = `
Array [
  Object {
    "_id": Any<ObjectId>,
    "siren": "038995593",
  },
  Object {
    "_id": Any<ObjectId>,
    "siren": "039084561",
  },
]
`;

exports[`SireneStockUniteLegaleCli parse should persist sirene data 1`] = `
Array [
  Object {
    "_id": Any<ObjectId>,
    "activitePrincipaleUniteLegale": "86.90F",
    "anneeCategorieEntreprise": "",
    "anneeEffectifsUniteLegale": "",
    "caractereEmployeurUniteLegale": "",
    "categorieEntreprise": "",
    "categorieJuridiqueUniteLegale": "9210",
    "dateCreationUniteLegale": "1987-10-28",
    "dateDebut": "2012-08-01",
    "dateDernierTraitementUniteLegale": "2024-03-22 14:26:06",
    "denominationUniteLegale": "CENTRE MEDICO PSYCHO PEDAGOGIQUE",
    "denominationUsuelle1UniteLegale": "",
    "denominationUsuelle2UniteLegale": "",
    "denominationUsuelle3UniteLegale": "",
    "economieSocialeSolidaireUniteLegale": "",
    "etatAdministratifUniteLegale": "A",
    "identifiantAssociationUniteLegale": "",
    "nicSiegeUniteLegale": "00035",
    "nomUniteLegale": "",
    "nomUsageUniteLegale": "",
    "nombrePeriodesUniteLegale": "6",
    "nomenclatureActivitePrincipaleUniteLegale": "NAFRev2",
    "prenom1UniteLegale": "",
    "prenom2UniteLegale": "",
    "prenom3UniteLegale": "",
    "prenom4UniteLegale": "",
    "prenomUsuelUniteLegale": "",
    "pseudonymeUniteLegale": "",
    "sexeUniteLegale": "",
    "sigleUniteLegale": "",
    "siren": Object {
      "siren": "300483146",
    },
    "societeMissionUniteLegale": "",
    "statutDiffusionUniteLegale": "O",
    "trancheEffectifsUniteLegale": "",
    "unitePurgeeUniteLegale": "",
  },
  Object {
    "_id": Any<ObjectId>,
    "activitePrincipaleUniteLegale": "55.20Z",
    "anneeCategorieEntreprise": "",
    "anneeEffectifsUniteLegale": "",
    "caractereEmployeurUniteLegale": "",
    "categorieEntreprise": "",
    "categorieJuridiqueUniteLegale": "9210",
    "dateCreationUniteLegale": "1900-01-01",
    "dateDebut": "2008-01-01",
    "dateDernierTraitementUniteLegale": "2024-03-22 14:26:06",
    "denominationUniteLegale": "EDUCATION POPULAIRE DU FAUBOURG VOSGES",
    "denominationUsuelle1UniteLegale": "",
    "denominationUsuelle2UniteLegale": "",
    "denominationUsuelle3UniteLegale": "",
    "economieSocialeSolidaireUniteLegale": "",
    "etatAdministratifUniteLegale": "A",
    "identifiantAssociationUniteLegale": "",
    "nicSiegeUniteLegale": "00016",
    "nomUniteLegale": "",
    "nomUsageUniteLegale": "",
    "nombrePeriodesUniteLegale": "3",
    "nomenclatureActivitePrincipaleUniteLegale": "NAFRev2",
    "prenom1UniteLegale": "",
    "prenom2UniteLegale": "",
    "prenom3UniteLegale": "",
    "prenom4UniteLegale": "",
    "prenomUsuelUniteLegale": "",
    "pseudonymeUniteLegale": "",
    "sexeUniteLegale": "",
    "sigleUniteLegale": "",
    "siren": Object {
      "siren": "300846177",
    },
    "societeMissionUniteLegale": "",
    "statutDiffusionUniteLegale": "O",
    "trancheEffectifsUniteLegale": "",
    "unitePurgeeUniteLegale": "",
  },
]
`;
